internal/pkg/cli/task_run.go

```diff
@@ -0,0 +1,242 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/config"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/docker/dockerfile"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type runTaskVars struct {
+	*GlobalOpts
+	Num    uint8
+	CPU    uint16
+	Memory uint16
+
+	Image          string
+	DockerfilePath string
+
+	TaskRole string
+
+	SubnetID         string
+	SecurityGroupIDs []string
+	App              string
+	Env              string
+
+	EnvVars  map[string]string
+	Commands string
+}
+
+type runTaskOpts struct {
+	runTaskVars
+
+	// Interfaces to interact with dependencies.
+	fs    afero.Fs
+	store store
+	df    dockerfileParser
+
+	// sets up Dockerfile parser using fs and input path
+	setupParser func(opts *runTaskOpts)
+}
+
+func newTaskRunOpts(vars runTaskVars) (*runTaskOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("new config store: %w", err)
+	}
+
+	return &runTaskOpts{
+		runTaskVars: vars,
+
+		fs:    &afero.Afero{Fs: afero.NewOsFs()},
+		store: store,
+
+		setupParser: func(o *runTaskOpts) {
+			o.df = dockerfile.New(o.fs, o.DockerfilePath)
+		},
+	}, nil
+}
+
+// Validate returns an error if the flag values passed by the user are invalid.
+func (o *runTaskOpts) Validate() error {
+	if o.Num == 0 {
+		return errors.New("number of tasks must be positive")
+	}
+
+	if o.CPU == 0 {
+		return errors.New("CPU units must be positive")
+	}
+
+	if o.Memory == 0 {
+		return errors.New("memory must be positive")
+	}
+
+	if o.Image != "" && o.DockerfilePath != "" {
+		return errors.New("cannot specify both image and dockerfile path")
+	}
+
+	if o.Image != "" {
+		if err := o.validateImageName(); err != nil {
+			return err
+		}
+	}
+
+	if o.DockerfilePath != "" {
+		if _, err := o.fs.Stat(o.DockerfilePath); err != nil {
+			return err
+		}
+	}
+
+	if o.Env != "" && (o.SubnetID != "" || o.SecurityGroupIDs != nil) {
+		return errors.New("can only specify one of a)env and b)subnet id and (or) security groups")
+	}
+
+	if o.SubnetID != "" {
+		if err := o.validateSubnetID(); err != nil {
+			return err
+		}
+	}
+
+	if o.SecurityGroupIDs != nil {
+		if err := o.validateSecurityGroupIDs(); err != nil {
+			return err
+		}
+	}
+
+	if o.App != "" {
+		if err := o.validateAppName(); err != nil {
+			return err
+		}
+	}
+
+	if o.Env != "" {
+		if err := o.validateEnvName(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *runTaskOpts) validateImageName() error {
+	valid, err := regexp.MatchString(`^\d+\.dkr\.ecr\.[a-z0-9\-]+.amazonaws.com/[a-z][a-z0-9\-]*$`, o.Image)
+	if err != nil {
+		return fmt.Errorf("validate image name: %w", err)
+	}
+
+	if !valid {
+		return errors.New("image name is malformed")
+	}
+
+	return nil
+}
+
+func (o *runTaskOpts) validateSubnetID() error {
```

Comment:
this method is a good candidate for moving to validate.go--that way, you can reuse it as a validator function when you call prompt.Get() in askSubnetId()

Reply:
I will probably not validate subnet id and security group ids because that seems to be an overkill (agreeing with what Penghao said). But great suggestion! I will move validateImageName to validate.go if I am still keeping it (haven't decided yet)